src/xenia/hid/winkey/winkey_input_driver.cc

```diff
@@ -9,16 +9,96 @@
 
 #include "xenia/hid/winkey/winkey_input_driver.h"
 
+#include "xenia/base/logging.h"
 #include "xenia/base/platform_win.h"
 #include "xenia/hid/hid_flags.h"
 #include "xenia/hid/input_system.h"
 #include "xenia/ui/virtual_key.h"
 #include "xenia/ui/window.h"
 
+#define DEFINE_keybind(name, default_value, description) \
+  DEFINE_string(name, default_value,                     \
+                "List of keys to bind to " description   \
+                ", separated by spaces",                 \
+                "KeyBind")
+
+// clang-format off
+DEFINE_keybind(keybind_dpad_left        , " ^A   ", "DPAD_LEFT"          );
+DEFINE_keybind(keybind_dpad_right       , " ^D   ", "DPAD_RIGHT"         );
+DEFINE_keybind(keybind_dpad_down        , " ^S   ", "DPAD_DOWN"          );
+DEFINE_keybind(keybind_dpad_up          , " ^W   ", "DPAD_UP"            );
+DEFINE_keybind(keybind_left_thumb_left  , " _A   ", "LEFT_THUMB_LEFT"    );
+DEFINE_keybind(keybind_left_thumb_right , " _D   ", "LEFT_THUMB_RIGHT"   );
+DEFINE_keybind(keybind_left_thumb_down  , " _S   ", "LEFT_THUMB_DOWN"    );
+DEFINE_keybind(keybind_left_thumb_up    , " _W   ", "LEFT_THUMB_UP"      );
+DEFINE_keybind(keybind_left_thumb       , " F    ", "LEFT_THUMB_PRESSED" );
+DEFINE_keybind(keybind_right_thumb_up   , " 0x26 ", "RIGHT_THUMB_UP"     );
+DEFINE_keybind(keybind_right_thumb_down , " 0x28 ", "RIGHT_THUMB_DOWN"   );
+DEFINE_keybind(keybind_right_thumb_right, " 0x27 ", "RIGHT_THUMB_RIGHT"  );
+DEFINE_keybind(keybind_right_thumb_left , " 0x25 ", "RIGHT_THUMB_LEFT"   );
+DEFINE_keybind(keybind_right_thumb      , " K    ", "RIGHT_THUMB_PRESSED");
+DEFINE_keybind(keybind_x                , " L    ", "X"                  );
+DEFINE_keybind(keybind_b                , " 0xDE ", "B"                  );
+DEFINE_keybind(keybind_a                , " 0xBA ", "A"                  );
+DEFINE_keybind(keybind_y                , " P    ", "Y"                  );
+DEFINE_keybind(keybind_left_trigger     , " Q I  ", "LEFT_TRIGGER"       );
+DEFINE_keybind(keybind_right_trigger    , " E O  ", "RIGHT_TRIGGER"      );
+DEFINE_keybind(keybind_back             , " Z    ", "BACK"               );
+DEFINE_keybind(keybind_start            , " X    ", "START"              );
+DEFINE_keybind(keybind_left_shoulder    , " 1    ", "LEFT_SHOULDER"      );
+DEFINE_keybind(keybind_right_shoulder   , " 3    ", "RIGHT_SHOULDER"     );
+// clang-format on
+
+#define IS_KEY_TOGGLED(key) ((GetKeyState(key) & 0x1) == 0x1)
+#define IS_KEY_DOWN(key) ((GetAsyncKeyState(key) & 0x8000) == 0x8000)
+
 namespace xe {
 namespace hid {
 namespace winkey {
 
+void WinKeyInputDriver::ParseKeyBinding(ui::VirtualKey controllerKey,
```

Comment:
Please clean up your variable names according to [the style guide](https://google.github.io/styleguide/cppguide.html#Variable_Names).